src/library_async.js

```diff
@@ -341,9 +424,11 @@ mergeInto(LibraryManager.library, {
     },
   },
 
+  emscripten_sleep__sig: 'vi', // TODO: others
   emscripten_sleep__deps: ['$safeSetTimeout'],
   emscripten_sleep: function(ms) {
-    Asyncify.handleSleep((wakeUp) => safeSetTimeout(wakeUp, ms));
+    // TODO: add return in the others.
```

Comment:
What others?

Reply:
All other suspending methods. I'm adding a better comment now. Basically, for the new API to work sleep etc. all need to have a sig, and to return a value (the Promise).